test/ad/distributions.jl

```diff
@@ -18,14 +18,18 @@
     to_posdef_diagonal(a::AbstractVector) = Diagonal(a.^2 .+ 1)
 
     # Define adjoints for Tracker
-    to_posdef(A::TrackedMatrix) = Tracker.track(to_posdef, A)
-    Tracker.@grad function to_posdef(A::TrackedMatrix)
-        data_A = Tracker.data(A)
-        S = data_A * data_A' + I
-        function pullback(∇)
-            return ((∇ + ∇') * data_A,)
+    if AD == "All" || AD == "ForwardDiff_Tracker"
+        @eval begin
+            to_posdef(A::TrackedMatrix) = Tracker.track(to_posdef, A)
+            Tracker.@grad function to_posdef(A::TrackedMatrix)
+                data_A = Tracker.data(A)
+                S = data_A * data_A' + I
+                function pullback(∇)
+                    return ((∇ + ∇') * data_A,)
+                end
+                return S, pullback
+            end
```

Comment:
Why is `@eval` needed here but not below?

Reply:
This is a mistake and actually caused a nasty bug that I was hunting for a while now. Thanks for pointing this out. Is it even possible to overload a function within an if condition somehow?

Reply:
I assume moving the function definitions to a separate file and including it conditionally should work?